### PR TITLE
refactor: completely rewrite the logger

### PR DESCRIPTION
**Pull request by SukkaW**
Since hexo-bunyan is too heavy and hard to maintained, I have rewritten a new logger with nearly the same behavior (except `debug.log` file output). ~~So far the unit test is not finished yet.~~ Coverage is now 100%.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,7 +27,6 @@ Option | Description | Default
 --- | --- | ---
 `debug` | Display debug message and save log to `debug.log` file. | `false`
 `silent` | Don't display message in console. | `false`
-`name` | Name | `hexo`
 
 ## License
```
Comment (YoshinoriN) [inline README.md:32]: Please delete line number 35

```
[bunyan]: https://github.com/trentm/node-bunyan
```

File: lib/log.js [hunk 2]
```diff
@@ -22,77 +28,89 @@ const levelColors = {
...
+    this._debug = options.debug || false;
 
-require('util').inherits(ConsoleStream, Writable);
+    this.level = INFO;
 
-ConsoleStream.prototype._write = function(data, enc, callback) {
-  const level = data.level;
-  let msg = '';
+    if (silent) {
+      this.level = WARN;
```
Comment (YoshinoriN) [inline lib/log.js:45]: Was this done intentionally? 
Even if the `silent` option is true, `hexo-log` display a log message higher than the `WARN` level log.

Comment (SukkaW) [reply lib/log.js:45]: Oh I forgot the behavior of `hexo-log`. `hexo-log` won't display any message if `silent` is true.

Comment (SukkaW) [reply lib/log.js]: Update: now the logger won't output anything if `silent` is set to `true`.

**Pull request merged**